lib/kubernetes-deploy/kubernetes_resource/daemon_set.rb

```diff
@@ -17,8 +18,11 @@ def status
 
     def deploy_succeeded?
       return false unless exists?
-      rollout_data["desiredNumberScheduled"].to_i == rollout_data["updatedNumberScheduled"].to_i &&
-      rollout_data["desiredNumberScheduled"].to_i == rollout_data["numberReady"].to_i &&
+
+      considered_pods = @pods.select { |p| @nodes.map(&:name).include?(p.node_name) }
+      (rollout_data["desiredNumberScheduled"].to_i == rollout_data["updatedNumberScheduled"].to_i &&
```

Comment:
ok so i am well aware this multi-line complicated conditional isn't the prettiest thing in the world, but hey rubocop allowed it!

Open to ideas on how to make this clearer if the boolean logic is getting to sophisticated.